### ISSUE TITLE
[1.x] Templatizer leaks instance <template>  element and its dataHost onto prototype

### ISSUE THREAD

**Issue by kevinpschaaf** [Polymer/polymer#5096]
### Description
The first time a template is "templatized", the reference to the `<template>` element instance is stored on the Templatized constructor's prototype, which is cached on the cached template content for any nested templates.  This causes a leak not only of the template, but its `dataHost` reference, inadvertently causing a leak of potentially expensive elements all the way up the tree.  The assumption is that once an element is detached and gc'ed, all `dataHost` references back up the tree to it are also gc'ed.  This leak via the `<template>` element prevents proper gc from occurring.

#### Live Demo
http://jsbin.com/remilab/edit?html,console,output

#### Steps to Reproduce
Open jsBin

#### Expected Results
No reference to the template's dataHost is retained

#### Actual Results
After being removed, the `instance` element is retained via 
```js
    Polymer
      .telemetry
      .registrations
      .slice(-1)[0]
      ._notes[0]
      .templateContent
      ._ctor
      .prototype
      ._template
      .dataHost
```

### Browsers Affected

All

### Versions
- Polymer: v1.x

**Comment by kevinpschaaf** [comment]
The offending line looks to be here: https://github.com/Polymer/polymer/blob/1.x/src/lib/template/templatizer.html#L201

2.x does not appear to suffer from the same leak, as the `<template>` instance is stored on a subclass of the cached Templatized constructor each call to `templatize`, rather than the cached one directly: https://github.com/Polymer/polymer/blob/master/lib/utils/templatize.html#L518-L522

**Comment by TimvdLippe** [comment]
@kevinpschaaf I have updated your links to use permalinks to prevent linkrot and have nice in-lining of the corresponding source code

**Comment by TimvdLippe** [comment]
This has been fixed in #5102
